parquet-column/src/main/java/org/apache/parquet/schema/Types.java

```diff
@@ -406,6 +408,11 @@ protected PrimitiveType build(String name) {
                   meta.getPrecision() <= MAX_PRECISION_INT64,
                   "INT64 cannot store " + meta.getPrecision() + " digits " +
                   "(max " + MAX_PRECISION_INT64 + ")");
+              if (meta.getPrecision() <= MAX_PRECISION_INT32) {
+                String msg = String.format("Consider use INT32 instead of INT64 for storing %d digits; see %s for details.",
```

Comment:
SLF4J can format the message instead of using `String.format`. Minor, but it would help with consistency.

Also, I think the error message should be more direct: `Log.warn("Decimal with {} digits is stored in an INT64, but fits in an INT32. See {}.", precision, url)`